kitsune/wiki/strategies.py

```diff
@@ -296,6 +300,49 @@ def translate(self, l10n_request: TranslationRequest) -> TranslationResult:
         result.method = TranslationMethod.HYBRID
         return result
 
+    def reject_stale_translations(self) -> list[Revision]:
+        """
+        Reject stale, unreviewed machine translations.
+        """
+        if not settings.HYBRID_ENABLED_LOCALES:
+            return []
+
+        results = []
+        for rev in self.content_manager.get_stale_translations(
+            locales=settings.HYBRID_ENABLED_LOCALES
+        ):
+            self.content_manager.reject_revision(rev, comment="No longer relevant.")
+            results.append(rev)
+        return results
+
+    def publish_unreviewed_within_grace_period(self) -> list[Revision]:
+        """
+        Publish fresh machine translations that have not been reviewed within the grace period.
+        """
+        if not (settings.HYBRID_REVIEW_GRACE_PERIOD and settings.HYBRID_ENABLED_LOCALES):
+            return []
+
+        results = []
+        for rev in self.content_manager.get_fresh_translations(
+            locales=settings.HYBRID_ENABLED_LOCALES,
+            created__lt=Now() - timedelta(days=settings.HYBRID_REVIEW_GRACE_PERIOD),
+        ):
+            self.content_manager.publish_revision(
+                rev,
+                comment=(
+                    "Automatically approved because it was not reviewed within"
+                    f" {settings.HYBRID_REVIEW_GRACE_PERIOD} day(s)."
+                ),
+            )
+            results.append(rev)
+        return results
+
+    def handle_pending_reviews(self, asynchronous=True):
+        if asynchronous:
+            handle_pending_reviews_task.delay()
+        self.reject_stale_translations()
```

Comment:
Do we need to do this in bulk through Celery?
I was thinking that rejection could be part of the revision creation. When a new revision is saved we could check there if we need to invalidate anything and just call the manager methods and helpers. This should greatly simplify the implementation and offloads the revision management to the manager.
Eg
- New parent revision -> invalidate pending translations 
- New revision on the translation -> invalidate pending translation